### ISSUE TITLE
# does not disappear from notification icon once notifications are seen

### ISSUE THREAD

**Issue by deutrino** [stom79/mastalab#116]
Steps to reproduce:

1. Load the home timeline on an account with some notifications. 
2. Switch to the notifications timeline. 
3. Switch back to the home timeline

What actually happens:

Typically, the number of unread notifications which is superimposed on the notification timeline icon stays the same. 

What I expect to happen:

Once I've seen the notifications (and maybe switched away from the notifications timeline), the number should disappear. 

Mastalab version: 1.6.5.1

**Comment by stom79** [comment]
In fact, you need to click on the tab to remove it.

> Once I've seen the notifications (and maybe switched away from the notifications timeline), the number should disappear.

I remember I got some troubles when I wanted to implement that. I will try again.

**Comment by deutrino** [comment]
The number seems to reflect reality less nowadays? Like sometimes it seems like it will change when no new interactions have appeared.

**Comment by stom79** [comment]
Yes. I have to work on this behavior. It works with a bookmark which is changed when the id of the displayed toot is greater than its current value.
One difficult part is to say: "This toot has been read => updates the bookmark", because how to know if it was read?
A new toot can be displayed without being visible. Example: I am on notifications tab, I receive new toots in the home timeline, I don't read them and I close the app. I can't update the bookmark. Also I receive a push notification, do I need to update this bookmark? I am at the middle of the home timeline  and I receive new toots at the top (they are displayed but not visible) and I close the app without reading them. Etc.

**Comment by deutrino** [comment]
I've gotten used to tap-to-dismiss on this but the numbers are frequently inaccurate and don't always show when new activity has happened. I would say I notice this several times per week.
